finat/quadrature_element.py

```diff
@@ -127,6 +123,19 @@ def basis_evaluation(self, order, ps, entity=None, coordinate_mapping=None):
     def point_evaluation(self, order, refcoords, entity=None):
         raise NotImplementedError("QuadratureElement cannot do point evaluation!")
 
+    @property
+    def dual_basis(self):
+        ps = self._rule.point_set
+        Q = gem.Literal([self._rule.weights])
```

Comment:
> Do you need to rank-extend the weights when making `Q`, I guess yes, because the first dimension of `Q` is the number of functionals?

Yes, otherwise I could just use `Q = self._rule.weight_expression` I think

Reply:
> I think that if this a `TensorProductQuadratureRule` then `self._rule.weights` doesn't exist.

Hmm. Possible design mistake somewhere?

Reply:
edit: I mean design mistake by me!

Reply:
For now added `NotImplementedError`